doc/develop/west/alias.rst

```diff
@@ -0,0 +1,55 @@
+.. _west-aliases:
+
+West aliases
+############
+
+West allows to add alias commands to the local, global or system configuration files.
+These aliases make it easy to add shortcuts for frequently used, or hard to memorize
+commands for ease of development.
+
+Similar to how ``git`` aliases work, the alias command is replaced with the alias'
+full text and parsed as a new shell argument list. This enables adding argument
+parameters as they were passed to the original command.
+
+To add a new alias simply call the ``west config`` command:
+
+.. code-block:: console
+
+   west config alias.command "other-command --with-arg"
+
+Recursion
+---------
+
+Recursive aliases are allowed as an alias command can contain other aliases, effectively
+building more complex, but easy to remember commands.
+
+Overriding
+----------
```

Comment:
Nit: these subsection names feel overkill; the first one has only one sentence. A blank line is probably enough.